src/dev/flang/ast/AbstractFeature.java

```diff
@@ -490,14 +490,16 @@ else if (this.compareTo(Types.resolved.f_choice) == 0)
             if (CHECKS) check
               (Errors.any() || p.calledFeature() != null);
 
-            if (p.calledFeature().isBaseChoice())
+            if (p.calledFeature().isChoice())
               {
                 if (lastP != null)
                   {
                     AstErrors.repeatedInheritanceOfChoice(p.pos(), lastP.pos());
                   }
                 lastP = p;
+                // NYI: BUG: we need sth. like hand down to figure out the choice generics.
```

Comment:
It is not sufficient anymore to just use actualTypeParameters of `p`. Need to change this

Reply:
would be good to have an example that does not work, at least in the comment. Even better would be to fix it :-)...

Reply:
@fridis sorry for not being clearer. previously the choice generics where just taken from the inherited choice feature.
e.g.
`option(T type): choice T nil is`

but now we can have e.g.:
`a : option String is`
To figure out the choice generics of a we need to hand down `String` and find the "one" feature that inherits the choice and take those generics.
This will need to be implemented in another PR.